grammar.js

```diff
@@ -277,7 +277,8 @@ module.exports = grammar({
     string: $ => seq(
       '"',
       repeat(choice(
-        /[^"`$]+/,
+        /[^"`$\\]+/,
+        seq('\\', /./), // escaped character
```

Comment:
There's one adjustment we need to make for performance reasons: A string with escape sequences can be recognized by the lexer all at once - as one single token. This is advantageous because the lexer is just faster than the parser, but also because it avoids allocating an unnecessary syntax node for each character.

There are two ways to ensure that the whole thing gets processed as a single token.
1. You can express the entire thing with one big RegExp.
2. You can wrap the rule in a call to `token`:

```js
_string_content: $ => token(repeat(choice(
  /^["`$]/,
  seq('\\', /./)
)))
```